src/changelogs/parse_changelogs.py

```diff
@@ -36,56 +34,166 @@ def run(self, version, logs):
+        New Items doesn't need to be a tag at all, etc
+        """
+
+        # Remove ' Changes' suffix, i.e. 'Hero Changes' -> 'Heroes'
+        heading = heading.replace(' Changes', '')
+
+        map = {
+            'Hero Gameplay': 'Heroes',
+            'Hero Gamepla': 'Heroes',
+            'Hero': 'Heroes',
+            'Item Gameplay': 'Items',
+            'New Items': 'Items',
+            'Misc Gameplay': self.default_tag,
+            'General': self.default_tag,
+            'General Change': self.default_tag,
+        }
 
-                    changelog_dict[resource_type][resource_name].append(line)
+        # headings in this list are not converted to tags
+        headings_to_ignore = ['Ranked Mode']
+        if heading in headings_to_ignore:
+            return None
 
-            if not resource_found:
-                changelog_dict[current_heading]['General'].append(line)
+        return map.get(heading, heading)
 
-        changelog_with_icons = self._embed_icons(changelog_dict)
-        os.makedirs(self.OUTPUT_CHANGELOGS, exist_ok=True)
-        json_utils.write(self.OUTPUT_CHANGELOGS + f'/date/{version}.json', changelog_with_icons)
-        return changelog_with_icons
+    def _register_tag(self, tags, tag, is_group_tag=True):
+        """
+        Registers a tag to the changelog's current unique tags,
+        and to the static unique list of tags.
+        """
+
+        if tag not in tags:
+            tags.append(tag)
+
+        if tag not in self.unique_tags:
+            self.unique_tags.append(tag)
+
+        if is_group_tag and tag not in self.unique_tag_groups:
+            self.unique_tag_groups.append(tag)
+
+        return tags
 
-    # mass find and replace of any resource names with the ability icon template
+    # mass find and replace of referenced tags to {{PageRef|tag}}
     def _embed_icons(self, changelog):
-        for header, log_groups in changelog.items():
-            for group_name, logs in log_groups.items():
-                for index, log in enumerate(logs):
-                    for resource_key in self.resources:
-                        resource = self.resources[resource_key]
-                        resource_name = resource['Name']
-                        if resource_name is None:
-                            continue
-
-                        if resource_name not in log:
-                            continue
-                        icon = f'{{AbilityIcon|Ability={resource_name}|Size=20}}'
-                        new_log = log.replace(resource_name, f'{icon} {resource_name}')
-                        changelog[header][group_name][index] = new_log
+        new_changelog = changelog.copy()
+        for index, log in enumerate(changelog):
+            tags = log['Tags']
+            description = log['Description']
+            remaining_description = description
+            for tag in tags:
+                template = 'PageRef'
```

Comment:
With this pageref system, how does the wiki side figure out what, for example, what the link is for "Siphon Life"? Does it search through every hero?

Reply:
Its still a bit up in the air right now, but ideally it doesn't need to know what type of resource it is. It just defaults to creating a link to [[Siphon Life]] and uses icon [[File:Siphon Life.svg]] or [[File:Siphon Life.png]]. This reduces the character count for each call drastically, which is important because editing changelogs on the frontend needs to be as streamlined and efficient as possible. Having only 1 parameter is the best way to accomplish that imo.

For now there's some logic on [[Module:PageRef]] that searches hero names and appends _MM to the filename, but this may get dropped down the line. I am planning on a PR designed for parsing icons from the files, so it will be revisited at that time.

Reply:
But c reating a link won't link it to the page name, ie. `[[Siphon Life]]` needs to link to `Abrams/SIphon Life`

I'd suggest having an identifier, eg. `Item#Basic Mag`, `Ability#Siphon Life`. This removes the potential for conflicts, and helps narrow the search scope

Reply:
It will go directly to [[Siphon Life]] where the individual ability page will be, rather than the hero page [[Abrams#thing to Siphon Life]].

We could consider an identifier anyways down the line, though considering their wiki links won't be identifier-specific, it seems moot. For example, the earlier discussion on [[Spirit Lifesteal]] the stat vs the item. My ideal world would be to have [[Item/Spirit Lifesteal]] and [[Stat/Spirit Lifesteal]] instead, following OOP principles. Though, I know it seems too daunting to most contributors and would probably be opposed, so I don't think its worth the effort. It would also require revisiting the OOP restructure for the backend, not just the frontend.

Reply:
Oh I thought we were doing ability pages like `<heroname>/<abilityname>`. If not, then that should in theory be okay.

While rare, I think naming conflicts does have potential to happen since the game is only going to get bigger. Eg. if they name an ability the same as an item, what would your solution be?

Reply:
@Surxe comment still outstanding. did we discuss this in a chat? Can't remember

Reply:
> @Surxe comment still outstanding. did we discuss this in a chat? Can't remember

Was buggin out on mobile earlier, wouldn't let me reply directly. You can see my reply here https://github.com/deadlock-wiki/deadbot/pull/84#issuecomment-2450861376